orttraining/orttraining/core/optimizer/graph_transformer_utils.cc

```diff
@@ -64,7 +65,13 @@ std::vector<std::unique_ptr<GraphTransformer>> GeneratePreTrainingTransformers(
     const std::unordered_set<std::string>& weights_to_train,
     const TrainingGraphTransformerConfiguration& config,
     const IExecutionProvider& execution_provider,
-    const std::unordered_set<std::string>& rules_and_transformers_to_disable) {
+    const std::unordered_set<std::string>& rules_and_transformers_to_disable,
+    const OrtDevice& device) {
```

Comment:
I don't think this device input is needed, see commands above.

Reply:
We want to perform some of the graph transformations only when the device is either cuda/rocm. And since we don't have the execution providers information yet, we could benefit from passing in the device information that is available on the `ORTModule` frontend side. Please let me know if you have alternate suggestions here.

Reply:
For "We want to perform some of the graph transformations only when the device is either cuda/rocm", do you mean we only implemented the precompute logic for CUDA/ROCM kernels? If in this case, what if the model mostly run on GPUs, but some Gather runs on CPU kernel, this might be possible when we look at some shape retrieving cases.